### PR TITLE
Simplify unstructured decoding process

### PR DESCRIPTION
**Pull request by billiford**
Many functions were relying on manifests to be of type `[]map[string]interface{}` and then processing these manifests by converting them to unstructured pointers and then assigning the return value to the `Object` of this pointer, which was of type `map[string]interface{}`.

Instead, we now convert all manifests to `unstructured.Unstructured` at the start of a `deployManifest` request. This makes other functions easier to work with, and they are no longer converting the manifests to unstructured then back again!

Additionally, we are now working without pointers unless functions "require" them. 

In short, this PR does:
- convert all manifests (`[]map[string]interface{}`) at the start of a `deployManifest` operation to `unstructured.Unstructured`
- simplifies our `mergeManifests` function in doing so
- simplifies our `kubernetes.SortManifests(...)` func to no longer return an error

I'd recommend reading the comment on the `kubernetes.ToUnstructured(...)` func as well, went down an interesting rabbit hole with what happens when we just assume the user has passed in a valid Kubernetes manifest and do no decoding.

### CODE REVIEW

**Comment by billiford** [comment]
I forked this off of #35, so we can merge this first or not... and deal with the consequences later.

**Review by edsonhor — APPROVED** [review]
Looks good.
